### PR TITLE
Feature/kak/add event type#910

### LINKED ISSUE #910: Create Event type in CMS
Similar to Destinations, create an Event type in the content management system/Django interface that can be associated with an existing Destination (though not required), and which has start and end dates.

### PR DESCRIPTION
**Pull request by flibbertigibbet**
## Overview

Adds `Event` type to CMS models and admin interface.

### Demo

cac_admin

Admin interface has JS widgets for picking dates/times:
cac_event_date_widget

Event date form validation:
date_validation

### Notes

Also updated/removed admin JS references based on third-party library changes.

## Testing Instructions

 * Run migrations
 * Visit admin site: http://localhost:8024/admin
 * Should be able to add and edit events
 * Should be able to perform actions from list view (publish or unpublish in bulk)

## Checklist
- [x] No gulp lint warnings
- [x] No python lint warnings
- [x] Python tests pass

Closes #910

### CODE REVIEW

File: python/cac_tripplanner/cac_tripplanner/settings.py [hunk 1]
```diff
@@ -259,9 +259,6 @@
 CKEDITOR_UPLOAD_PATH = 'uploads/'
 CKEDITOR_IMAGE_BACKEND = 'pillow'
 
-# TODO: delete later.
-CKEDITOR_JQUERY_URL = '//ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js'
```
[the remark this answers is not in the document]
Comment (flibbertigibbet) [reply python/cac_tripplanner/cac_tripplanner/settings.py]: `ckeditor` used to have a jQuery dependency; I think the `TODO` was recognition that the dependency was going to get removed, but at the time they hadn't cut a new release yet.

File: python/cac_tripplanner/destinations/models.py [hunk 2]
```diff
@@ -48,6 +57,32 @@ class Meta:
...
+    """Represents an event, which has a start and end date"""
+
+    class Meta:
+        ordering = ['priority', '-start_date']
+
+    name = models.CharField(max_length=50)
+    website_url = models.URLField(blank=True, null=True)
+    description = RichTextField()
+    start_date = models.DateTimeField()
+    end_date = models.DateTimeField()
```
[the remark this answers is not in the document]
Comment (flibbertigibbet) [reply python/cac_tripplanner/destinations/models.py]: Was thinking we would probably ignore the time portion of the date range on the front end, but it would be valuable to store in the case of a single-day event. Hours of operation can simply go in the description field, since we're not going to try to filter on that (or at least not yet).

File: python/cac_tripplanner/destinations/admin.py [hunk 3]
```diff
@@ -18,12 +18,8 @@ class DestinationAdmin(admin.OSMGeoAdmin):
     # Override map_template for custom address geocoding behavior
     map_template = 'admin/cac-geocoding-map.html'
 
-    # Override configurable URL for openlayers.js to support SSL
-    # default is: 'http://openlayers.org/api/2.13.1/OpenLayers.js'
-    openlayers_url = 'https://cdnjs.cloudflare.com/ajax/libs/openlayers/2.13.1/OpenLayers.js'
```
[the remark this answers is not in the document]
Comment (flibbertigibbet) [reply python/cac_tripplanner/destinations/admin.py]: Yup! The default changed to our override value.

**Comment by flibbertigibbet** [comment]
Regarding the name of the `destinations` app: yes, it occurred to me too that it looks silly now in the admin panel, now that there are non-destination "destinations." It didn't seem worth it to try to rename an entire Django app, though. I'll see if there's an easy way to override that string that shows in the admin panel as the app label. If not, I think it should be fine to leave that way; I don't think it would confuse our users too much.

Regarding the related destination delete button: yes, that is strange default behavior to offer that in the admin panel. Also weird that it offers to open a pop-up to directly add or edit destinations there, too. Not sure if it's worth it trying to override that behavior. If the user wants to simply unset the destination, there's an option for that at the top of the drop-down list. I'll also check if there's an easy override for the options on that widget.

**Comment by flibbertigibbet** [comment]
Setting a verbose name on the app changed the header in the main admin page content, but not the sidebar menu text.
image

**Comment by flibbertigibbet** [comment]
Found a way to override the default [related-field edit widget](https://github.com/django/django/blob/931c60c5216bd71bc11f489e00e063331cf21f40/django/contrib/admin/widgets.py#L237) to convince it to not display the add/edit/delete buttons. It was not an easy modification to discover, however, and in retrospect doubt it was worth the time spent.

This should be ready for another look.

File: python/cac_tripplanner/destinations/models.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (KlaasH) [reply python/cac_tripplanner/destinations/models.py]: Yeah, makes sense.
